emsdk/patches/0001-Fix-dyncall-and-dynamic-linking.patch

```diff
@@ -0,0 +1,49 @@
+From 7c7a8f37b79e357cc28cb44f99258c9841d7385a Mon Sep 17 00:00:00 2001
+From: Hood Chatham <roberthoodchatham@gmail.com>
+Date: Fri, 8 Apr 2022 15:54:49 -0700
+Subject: [PATCH] Fix dyncall and dynamic linking
+
+We don't compile with WASM_BIGINT so if there is a 64 bit integer
+in the dynamic call, Emscripten tries to call a legalizer wrapper.
+These legalizer wrappers are only generated in the main module so
+sometimes a side module tries to call a wrapper that doesn't exist.
+This adds an extra handler in that case.
```

Comment:
Please link to https://github.com/emscripten-core/emscripten/pull/16693 

Thanks for contributing this to Emscripten! It might be good to wait a bit until they are happy with the PR there before merging this.

Reply:
They suggested we should use `-sWASM_BIGINT` and shim `BigInt64Array` and `BigUint64Array` for Safari versions that don't have them. I think we can start out by using this patch and work on switching? Switching to `-sWASM_BIGINT` would be generally pretty good for us but it might take some effort.

Reply:
Opened #2643 to try to switch to the strategy recommended upstream.

Reply:
Hmm this patch also seems like it might not be necessary -- it was needed for `pyo3_samples` to work but it is unclear that it is needed for cryptography.